spec/spec_helper.rb

```diff
@@ -1,18 +1,137 @@
-require 'capybara'
-require 'capybara/rspec'
-require 'capybara/webkit'
+# use local 'lib' dir in include path
+$:.unshift File.dirname(__FILE__)+'/../lib'
+require 'json'
+require 'pp'
 require 'dotenv'
-require 'rspec/webservice_matchers'
+require 'gmail'
+require 'capybara/rspec'
+require 'capybara/poltergeist'
+
+Capybara.javascript_driver = :poltergeist
+Capybara.default_max_wait_time = 5
+
+RSpec.configure do |config|
+  config.color = true
+  config.order = :random
+end
 
 Dotenv.load
 
-Capybara.javascript_driver = :webkit
-Capybara.run_server        = false
+PASSWORD = 'salty pickles'.freeze
+EMAIL = ENV['EMAIL']
+EMAIL_PASSWORD = ENV['EMAIL_PASSWORD']
+PHONE = ENV['PHONE']
+if !EMAIL or !EMAIL_PASSWORD or !PHONE
+  abort("Must set EMAIL EMAIL_PASSWORD and PHONE env vars -- did you create a .env file?")
+end
+
+##############################################################################
+# helper methods
+
+def idp_signup_url
+  ENV['IDP_URL'] + '/sign_up/enter_email'
+end
+
+def sp_url
+  ENV['SP_URL']
+end
+
+def basic_auth_user
+  URI.parse(ENV['IDP_URL']).user
+end
+
+def basic_auth_pass
+  URI.parse(ENV['IDP_URL']).password
+end
 
-Capybara::Webkit.configure(&:allow_unknown_urls)
+def random_email_address
+  random_str = SecureRandom.hex(3)
+  EMAIL.dup.gsub(/@/, "+#{random_str}@")
+end
+
+def gmail
+  @_gmail ||= Gmail.connect!(EMAIL, EMAIL_PASSWORD)
+end
+
+def inbox_unread
+  gmail.inbox.emails(:unread)
+end
 
-module IM
-  def self.new_session
-    Capybara::Session.new(:webkit)
+def current_otp_from_email
+  inbox_unread.each do |email|
+    msg = email.message.parts[0].body
+    otp = msg.match(/(\d+) is your login.gov one-time passcode/)[1]
+    if otp
+      email.read!
+      return otp
+    end
   end
+  nil
+end
+
+def current_confirmation_link
+  inbox_unread.each do |email|
+    next unless email.subject == 'Confirm your email'
+    msg = email.message.parts[0].body
+    url = msg.match(/(https:.+confirmation_token=[\w\-]+)/)[1]
+    if url
+      email.read!
+      return url
+    end
+  end
+  nil
+end
+
+def check_for_otp
+  otp = nil
+  counter = 0
+  while (!otp) do
+    sleep 2
```

Comment:
yes the sleep is necessary to avoid spamming google. They can handle the load, but we want a sane way to time-out our attempts, so we sleep a little and then try again for `n` times.